gpiozero/boards.py

```diff
@@ -1185,3 +1185,97 @@ def on(self):
     def off(self):
         self.value = False
 
+class SevenSegmentDisplay(LEDBoard):
+    
+    def __init__(self, *pins, **kwargs):
+        # 7 segment displays must have 7 or 8 pins
+        if len(pins) < 7 or len(pins) > 8:
+            raise ValueError('SevenSegmentDisplay must have 7 or 8 pins')
+        # Don't allow 7 segments to contain collections
+        for pin in pins:
+            assert not isinstance(pin, LEDCollection)
+        pwm = kwargs.pop('pwm', False)
+        active_high = kwargs.pop('active_high', True)
+        initial_value = kwargs.pop('initial_value', False)
+        if kwargs:
+            raise TypeError('unexpected keyword argument: %s' % kwargs.popitem()[0])
+            
```

Comment:
Might be worth adding a code-comment in here that the LED segments in the `_layouts` array are listed in the order (a, b, c, d, e, f, g) ?